src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -244,12 +268,19 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         onnx_input_names = self.multitoken_engine.onnx_input_names_no_cache
         engine_input = self.tokens_to_engine_input(input_tokens, onnx_input_names)
 
-        if inputs.session_id is not None:
-            # if session_id is provided, we need to set it in engines
-            self.engine.session_id = inputs.session_id
-            self.multitoken_engine.session_id = inputs.session_id
-
         postprocessing_kwargs = dict(return_logits=inputs.return_logits)
+
+        if inputs.session_ids is not None:
+            session_ids = (
+                [inputs.session_ids]
+                if isinstance(inputs.session_ids, str)
+                else inputs.session_ids
+            )
+            # TODO: This is ugly, we should find a better way to pass
+            # session_id to the engine, maybe use the refactored pipeline
+            engine_input.append(numpy.array(session_ids))
```

Comment:
agreed, limitation of the "EnginePipeline" assumption - had thought we had the ability to do kwargs here for some reason...